src/controllers/controller.hpp

```diff
@@ -645,6 +656,19 @@ void Controller::execute_circuit(Circuit &circ,
       // Vector to store parallel thread output data
       std::vector<ExperimentResult> par_results(parallel_shots_);
       std::vector<std::string> error_msgs(parallel_shots_);
+
+    #ifdef _OPENMP
+    if (!parallel_nested_) {
+      if (parallel_shots_ > 1 && parallel_state_update_ > 1) {
+        // Nested parallel shots + state update
+        omp_set_nested(1);
+        result.metadata["omp_nested"] = true;
```

Comment:
It may be better to update `parallel_nested_` with `true`. But this is not mandatory.

Reply:
I didn't update it here since the variable was just needed to know if the outer loop was already parallelized or not. If it isn't each individual circuit execution has the option to be nested or not nested.